test/e2e/specs/format-api.test.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import {
+	clickBlockAppender,
+	getEditedPostContent,
+	META_KEY,
+	newPost,
+	pressWithModifier,
+} from '../support/utils';
+import { activatePlugin, deactivatePlugin } from '../support/plugins';
+
+/**
+ * The modifier keys needed to invoke a 'select the next word' keyboard shortcut.
+ *
+ * @type {string}
+ */
+const SELECT_WORD_MODIFIER_KEYS = process.platform === 'darwin' ? [ 'Shift', 'Alt' ] : [ 'Shift', 'Control' ];
+
+describe( 'Using Format API', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	afterAll( async () => {
+		await deactivatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	beforeEach( async () => {
+		await newPost();
+	} );
+
+	it( 'Format toolbar is present in a paragraph block', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'First paragraph' );
+		await page.mouse.move( 200, 300, { steps: 10 } );
+		expect( await page.$( '.editor-format-toolbar' ) ).not.toBeNull();
```

Comment:
What's does this test check that would not fail in the other? The format toolbar is always present with core formatting controls.

Reply:
Good point. It should probably actually check that the button itself exists in the toolbar, not that the toolbar exists. 

If I change it to check if the button itself exists then a failing first test would still cause a failing second test. So the first test would be slightly redundant - without it, the second test would still fail. However the first test would give more information about why it failed - because _the button doesn't exist_ as opposed to because _the button exists but the content wasn't properly marked up_.

Thoughts on whether the slight redundancy but more clarity is needed, or should the first test be removed altogether?

Reply:
As far as I understand, in the 2nd test you click on the button (`await page.click( '[aria-label="Custom Link"]' );`) so it has to be present in the UI otherwise the test will fail.

Regarding the point made by @iseulde, I would be okey with leaving this test if it would validate if HTML for the button renders as expected.

Reply:
Sorry, just to clarify, when you say:

>  I would be okey with leaving this test

You mean you'd be okay with removing it? Or leaving it in there?

Reply:
It can stay if it is updated to test the visuals to make it different than the 2nd test.